apm-lambda-extension/extension/route_handlers.go

```diff
@@ -99,8 +99,13 @@ func handleIntakeV2Events(agentDataChan chan AgentData) func(w http.ResponseWrit
 				Data:            rawBytes,
 				ContentEncoding: r.Header.Get("Content-Encoding"),
 			}
-			log.Println("Adding agent data to buffer to be sent to apm server")
-			agentDataChan <- agentData
+
+			select {
+			case agentDataChan <- agentData:
+				log.Println("Adding agent data to buffer to be sent to apm server")
+			default:
+				log.Println("Channel full : dropping event")
```

Comment:
```suggestion
				log.Println("Channel full: dropping event")
```

Reply:
Solved in [d9a8edc](https://github.com/elastic/apm-aws-lambda/pull/136/commits/d9a8edc0ca29cd924a0217ed22f9715a5c270c90)